src/main/java/io/wgj/ctci/Stack.java

```diff
@@ -1,21 +1,80 @@
 package io.wgj.ctci;
 
-public class Stack {
-    int[] stack;
+/**
+ * Stack is a stack data structure backed by a dynamic array. Stack can be
+ * initialized with a starting array size. It's written to develop familiarity
+ * with the stack data structure, and should otherwise not be used.
+ * Maximum allowed array is 1024. If you set a starting capacity larger than
+ * this, the maximum allowed array size is used as the starting capacity. An
+ * UnsupportedOperationException will be thrown when more the maximum allowed
+ * amount of elements are added.
+ */
+public class Stack<E> {
+    static final int MAX_ARRAY_SIZE = 1024;
+    static final int DEFAULT_CAPACITY = 10;
+    Object[] stack;
+    int size;
```

Comment:
Looking at your implementation, "size" should actually be called capacity and "position" should be called size.